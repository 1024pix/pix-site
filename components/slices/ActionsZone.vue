<template>
  <nav class="actions-zone">
    <locale-switcher></locale-switcher>
    <ul>
      <li v-for="(menuItem, index) in slice.items" :key="`item-${index}`">
        <nuxt-link
          :to="menuItem.link"
          class="actions-zone__item"
          :class="{
            'button button--with-border-focus': menuItem.style === 'button',
            link: menuItem.style === 'link',
          }"
        >
          <img
            v-if="menuItem.icon"
            :src="`/images/${menuItem.icon}`"
            class="actions-zone__item__icon"
            alt
            role="presentation"
            width="23"
            height="22"
          />
          {{ $prismic.asText(menuItem.name) }}
        </nuxt-link>
      </li>
    </ul>
  </nav>
</template>

<script>
export default {
  name: "SlicesActionsZone",
  props: {
    slice: {
      type: Object,
      default: null,
    },
  },
};
</script>

<style scoped lang="scss">
.actions-zone {
  display: flex;
  align-items: center;
  height: 100%;
  margin-left: 5px;

  ul {
    list-style: none;
    display: flex;
    padding-left: 0;

    @include device-is("tablet") {
      padding-left: 10px;
    }

    li::before {
      content: none;
    }
    li {
      align-self: center;
      display: flex;
    }
  }

  &__item {
    font-size: 0;
    color: $grey-60;
    font-family: $font-roboto;
    font-weight: $font-medium;
    letter-spacing: 0.008rem;
    line-height: 1.375rem;
    display: inline-flex;
    align-items: center;

    @include device-is("tablet") {
      font-size: 0.875rem;
    }

    &__icon {
      margin-right: 5px;
    }

    &.link:hover {
      color: $blue;
    }

    &.button {
      display: none;
    }
  }

  @include device-is("large-screen") {
    &__item {
      &.button {
        display: flex;
        color: $white;
        height: 36px;
      }
    }
  }
}
</style>
