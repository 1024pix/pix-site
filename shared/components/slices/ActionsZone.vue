<template>
  <nav class="actions-zone">
    <locale-switcher />
    <ul>
      <li v-if="slice.primary.helperLink" class="actions-zone__item-helper">
        <a :href="slice.primary.helperLink.url" class="actions-zone-item-helper__link">
          <img
            v-if="slice.primary.helperIcon"
            :src="`/images/${slice.primary.helperIcon}`"
            class="actions-zone-item-helper__icon"
            alt=""
            width="16"
            height="16"
          />
          {{ $prismic.asText(slice.primary.helperName) }}
        </a>
      </li>
      <li v-for="(menuItem, index) in slice.items" :key="`item-${index}`" class="actions-zone__item">
        <a
          :href="menuItem.link.url"
          class="actions-zone-item__link"
          :class="{
            'button button--with-border-focus': menuItem.style === 'button',
            link: menuItem.style === 'link',
          }"
        >
          <img
            v-if="menuItem.icon"
            :src="`/images/${menuItem.icon}`"
            class="actions-zone-item__icon"
            alt=""
            width="15"
            height="14"
          />
          {{ $prismic.asText(menuItem.name) }}
        </a>
      </li>
    </ul>
  </nav>
</template>

<script setup>
defineProps({
  slice: {
    type: Object,
    default: null,
  },
});
</script>

<style scoped lang="scss">
.actions-zone {
  display: flex;
  align-items: center;
  height: 100%;
  margin-left: 0.313rem;

  ul {
    list-style: none;
    display: flex;
    padding: 0;

    li::before {
      content: none;
    }
    li {
      align-self: center;
      display: flex;
    }
  }

  &__item-helper {
    padding: 0;

    &:hover {
      a {
        color: $blue;
      }
      img {
        filter: invert(30%) sepia(70%) saturate(1972%) hue-rotate(216deg) brightness(103%) contrast(101%);
      }
    }
  }
}

.actions-zone-item {
  &__link {
    font-size: 0;
    color: $grey-90;
    font-family: $font-roboto;
    font-weight: $font-medium;
    letter-spacing: 0.008rem;
    line-height: 1.375rem;
    display: inline-flex;
    align-items: center;
    padding: 0;
    border-radius: 0.25rem;

    &.button {
      display: none;
    }

    @include device-is('tablet') {
      padding: 0.5rem 1rem;
      font-size: 0.875rem;
      &.link {
        border: 1px solid $grey-60;

        img {
          margin-right: 0.313rem;
        }

        &:hover {
          color: $white;
          background-color: $grey-60;

          img {
            filter: brightness(0) invert(1);
          }
        }
        &:focus {
          color: $white;
          background-color: $grey-90;
          border: 1px solid $grey-90;
          outline: 1px solid $white;
          outline-offset: -4px;

          img {
            filter: brightness(0) invert(1);
          }
        }
      }
    }

    @include device-is('desktop') {
      &.button {
        display: flex;
        color: $white;
        height: 36px;
      }
    }
  }
}

.actions-zone-item-helper {
  &__link {
    display: none;
    font-size: 0.875rem;
    margin-right: 0.75rem;
    padding: 0 1.5rem;
    border-right: 1px solid $grey-20;
    color: $grey-70;
    font-family: $font-roboto;
    font-weight: $font-medium;

    @include device-is('tablet') {
      display: flex;
      align-items: center;
      height: 36px;
    }
  }

  &__icon {
    margin-right: 0.313rem;
  }
}
</style>
