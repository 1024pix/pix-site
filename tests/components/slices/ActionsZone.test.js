import { shallowMount } from '@vue/test-utils'
import ActionsZone from '~/components/slices/ActionsZone'

describe('Slice: Actions Zone', () => {
  let component

  it('it should display the locale switcher', () => {
    // given
    component = shallowMount(ActionsZone, {
      stubs: {
        'locale-switcher': true,
        'pix-prismic-link': true,
      },
      propsData: {
        slice: {
          items: [],
        },
      },
    })

    // when
    const result = component.html()

    // then
    expect(result).toContain('<locale-switcher-stub')
  })
})
