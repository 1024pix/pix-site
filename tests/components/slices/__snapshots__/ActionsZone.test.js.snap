// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slice: Actions Zone When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to false it should display the language switcher 1`] = `
"<nav class=\\"actions-zone\\">
  <language-switcher-stub type=\\"with-dropdown\\"></language-switcher-stub>
  <ul></ul>
</nav>"
`;

exports[`Slice: Actions Zone When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to true it should not display the language switcher 1`] = `
"<nav class=\\"actions-zone\\">
  <language-switcher-stub type=\\"with-dropdown\\"></language-switcher-stub>
  <ul></ul>
</nav>"
`;
